### PR TITLE
Test compilation for inference

### PR DESCRIPTION
**Pull request by kvantricht**
FYI @gabrieltseng

Test results of this compilation on 10,000 samples (pixels):

**without compilation:**
- warm-up time:  4s
- average inference time: 6s

**with compilation:**
- warm-up time: 37s
- average inference time: 1.75s

Preliminary conclusion: for small inference jobs, warm-up time of JIT compilation does not pay off and actually leads to an increase of inference time. However, when we'd succeed at caching the compiled (and warmed-up!) Presto encoder for larger inference jobs, a more than 3-fold reduction in inference time is observed which is quite significant. We will test this more at larger scale also to see if caching actually works, however a C++ compiler first needs to be installed in the openEO environment on CDSE before this can succeed.

### CODE REVIEW

**Comment by gabrieltseng** [comment]
Thanks for these benchmaks Kristof - very interesting

File: presto/dataset.py [hunk 1]
```diff
@@ -197,6 +197,23 @@ def split_df(
         return train, val
 
 
+class WorldCerealDataset(WorldCerealBase):
+    # The simplest pd.DataFrame based dataset just
+    # for inference without labels
+    def __getitem__(self, idx):
```
Comment (gabrieltseng) [inline presto/dataset.py:203]: nit - could this just replace the currently not implemented `__getitem__` function in `WorldCerealBase` to limit how many extensions of the base class there are?

**Review by gabrieltseng — APPROVED** [review]
One nit, otherwise LGTM :)

File: presto/dataset.py (hunk 1 above)
Comment (kvantricht) [reply presto/dataset.py]: That's a great suggestion! At some point all these datasets could anyway use a cleanup. But for now, moved `__getitem__` as suggested.

**Pull request merged**